config/initializers/statsd.rb

```diff
@@ -26,21 +26,42 @@ class << Samson
 end
 
 ActiveSupport::Notifications.subscribe("job_queue.samson") do |*, payload|
+  # deploys running/queued
+  deploys_metrics = payload.delete(:deploys)
+  deploys_metrics.each { |key, value| Samson.statsd.gauge "deploys.#{key}", value }
+
+  # total jobs queued/running reporting
   payload.each { |key, value| Samson.statsd.gauge "job.#{key}", value }
 end
 
+ActiveSupport::Notifications.subscribe("job_status.samson") do |*, payload|
+  Samson.statsd.increment "#{payload[:type]}.#{payload[:status]}"
+end
+
 ActiveSupport::Notifications.subscribe("system_stats.samson") do |*, payload|
   payload.each { |key, value| Samson.statsd.gauge key.to_s, value }
 end
 
+ActiveSupport::Notifications.subscribe("wait_for_build.samson") do |*args|
+  event = ActiveSupport::Notifications::Event.new(*args)
+  tags = [
+    "project:#{event.payload.fetch(:project)}",
+    "gcb:#{event.payload.fetch(:gcb)}"
+  ]
+
+  Samson.statsd.histogram "builds.time.wait_time", event.duration, tags: tags
```

Comment:
should report a timeing instead of a histogram ... or are they the same under the hood ?

Reply:
I think they are essentially the same thing under the hood going by the datadog docs